libs/blocks/preflight/checks/seo.js

```diff
@@ -246,6 +246,30 @@ function compareResults(result, link) {
   return true;
 }
 
+export async function validLinkFilter(area = document) {
+  const { preflight } = await getServiceConfig(window.location.origin);
+  const knownBadUrls = preflight?.ignoreDomains
+    ? preflight?.ignoreDomains.split(',').map((url) => url.trim())
+    : KNOWN_BAD_URLS;
+  return [...area.querySelectorAll('a')]
+  .filter((link) => {
+    if (
+      link.href // Has an href tag
+      && !link.href.includes('tel:')  // Is not a phone number
+      && !link.href.includes('mailto:')  // Is not an email address
+      && !link.href.startsWith('#') // Is not a local link
+      && !link.href.startsWith('https://#') // Is not a local link
+      && !link.href.includes('local') // Is not a local link
+      && !link.href.includes('bookmark://') // Ignore bookmarks
+      && !link.closest('.preflight') // Is not inside preflight
+      && !knownBadUrls.some((url) => url === link.hostname) // Is not a known bad url
+    ) {
+      return true;
+    }
+    return false;
```

Comment:
You should be able to just return this, no need for `if`
```suggestion
      return link.href // Has an href tag
      && !link.href.includes('tel:')  // Is not a phone number
      && !link.href.includes('mailto:')  // Is not an email address
      && !link.href.startsWith('#') // Is not a local link
      && !link.href.startsWith('https://#') // Is not a local link
      && !link.href.includes('local') // Is not a local link
      && !link.href.includes('bookmark://') // Ignore bookmarks
      && !link.closest('.preflight') // Is not inside preflight
      && !knownBadUrls.some((url) => url === link.hostname) // Is not a known bad url
```